src/Raven.Server/Monitoring/Snmp/Objects/Database/5.1/5.1.11/ActiveOngoingTasksBase.cs

```diff
@@ -0,0 +1,105 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Lextm.SharpSnmpLib;
+using Raven.Client.Documents.Subscriptions;
+using Raven.Client.Json.Serialization;
+using Raven.Client.ServerWide;
+using Raven.Server.ServerWide;
+using Raven.Server.ServerWide.Context;
+
+namespace Raven.Server.Monitoring.Snmp.Objects.Database;
+
+public abstract class ActiveOngoingTasksBase : DatabaseBase<Integer32>
+{
+    protected ActiveOngoingTasksBase(ServerStore serverStore, string dots)
+        : base(serverStore, dots)
+    {
+    }
+
+    protected override IEnumerable<RawDatabaseRecord> GetDatabases(TransactionOperationContext context)
+    {
+        foreach (var database in base.GetDatabases(context))
+        {
+            if (database.IsDisabled)
+                continue;
+
+            if (ServerStore.DatabasesLandlord.IsDatabaseLoaded(database.DatabaseName) == false)
+                continue;
+
+            yield return database;
+        }
+    }
+
+    protected abstract int GetCount(TransactionOperationContext context, RachisState rachisState, string nodeTag, RawDatabaseRecord database);
+
+    protected override Integer32 GetData()
+    {
+        var count = 0;
+        using (ServerStore.ContextPool.AllocateOperationContext(out TransactionOperationContext context))
+        using (context.OpenReadTransaction())
+        {
+            var rachisState = ServerStore.CurrentRachisState;
+            var nodeTag = ServerStore.NodeTag;
+
+            foreach (var database in GetDatabases(context))
+            {
+                count += GetCount(context, rachisState, nodeTag, database);
+            }
+        }
+
+        return new Integer32(count);
+    }
+
+    protected static int GetNumberOfActiveElasticSearchEtls(RachisState rachisState, string nodeTag, RawDatabaseRecord database) => CountTasks(rachisState, nodeTag, database.Topology, database.ElasticSearchEtls.Where(x => x.Disabled == false));
+
+    protected static int GetNumberOfActiveExternalReplications(RachisState rachisState, string nodeTag, RawDatabaseRecord database) => CountTasks(rachisState, nodeTag, database.Topology, database.ExternalReplications.Where(x => x.Disabled == false));
+
+    protected static int GetNumberOfActiveOlapEtls(RachisState rachisState, string nodeTag, RawDatabaseRecord database) => CountTasks(rachisState, nodeTag, database.Topology, database.OlapEtls.Where(x => x.Disabled == false));
+
+    protected static int GetNumberOfActivePeriodicBackups(RachisState rachisState, string nodeTag, RawDatabaseRecord database) => CountTasks(rachisState, nodeTag, database.Topology, database.PeriodicBackups.Where(x => x.Disabled == false));
+
+    protected static int GetNumberOfActiveQueueEtls(RachisState rachisState, string nodeTag, RawDatabaseRecord database) => CountTasks(rachisState, nodeTag, database.Topology, database.QueueEtls.Where(x => x.Disabled == false));
```

Comment:
Shouldn't we split it to Kafka and RabbitMQ ETLs? From user point of view those are separate ETL types

Reply:
For example on the cluster dashboard we have them separated

![image](https://github.com/ravendb/ravendb/assets/376551/0ac2145f-55fa-4fcf-a843-df0a8276017a)